module/zfs/dmu_objset.c

```diff
@@ -663,6 +663,9 @@ dmu_objset_own_impl(dsl_dataset_t *ds, dmu_objset_type_t type,
 		return (SET_ERROR(EINVAL));
 	} else if (!readonly && dsl_dataset_is_snapshot(ds)) {
 		return (SET_ERROR(EROFS));
+	} else if (!readonly && decrypt &&
+	    dsl_dir_bad_encryption_version(ds->ds_dir)) {
```

Comment:
nit: what do you think about using the `dmu_objset_*` version of the function here since we've just gotten the `objset_t` with `dmu_objset_from_ds()`.

Reply:
Not sure how I feel about that. Its easy enough to add a wrapper, but the same could be said for many other things, such as the call to `dsl_dataset_is_snapshot()` 3 lines above.

Reply:
Sure, I don't feel strongly about this I'm happy with whatever you prefer.